### DISCUSSION TITLE
Can't scroll up gVim terminal output

### DISCUSSION THREAD

**Discussion by Wanpaku** [vim/vim#15525] [General]
### Steps to reproduce

1. Manjaro OS
2. Install gVim with pacman.
3. Write any program with C++ or any other program language.

### Expected behaviour

Scrolling up and down with keyboards keys or mouse in the terminal output window.

### Version of Vim

gVim 9.1.0648-1

### Environment

OS Manjaro 24.0.6, terminal xfce4, value of echo $TERM is empty string.

### Logs and stack traces

```shell
I can’t scrolling up text in gVim terminal output window. It scrolling only downward after pressing appropriate keyboard keys. If output too big I can see only it’s bottom part. And the cursor changing to a cross in the terminal output window.
I don’t have such problem with usual textual Vim. And in others windows of gVim scrolling working fine.
How to fix it?
```

**Comment by chrisbra** [comment]
Really, this is a usage question here and therefore it should go to discussions and not an issue So have you checked the documentation of terminal mode?

**Reply by Wanpaku** [reply]
Yes, it didn't help.

**Reply by chrisbra** [reply]
Well, what did you try?

**Reply by Wanpaku** [reply]
I've tried to change terminal mode with Ctrl-\ and to set shell=/bin/xfce4-terminal.
Ctrl-\ just closing the terminal window. Set shell doesn't work for this parameters.

**Reply by chrisbra** [reply]
You should switch to Terminal normal mode and then you can navigate the window using the usual window scrolling commands.

Note: xfce4-terminal is not a shell, but a terminal. Try something like `bash` or `zsh`.

**Reply by Wanpaku** [reply]
Switching to Terminal normal mode doesn't working. It just closing the terminal window.
Also output of the command echo $TERM in gVim is an empty string.
In usual textual Vim this command has output  xterm_256color.
And in Vim I don't need to do any switching for scrolling terminal output. Why in gVim need I to switch terminal mode?

**Reply by chrisbra** [reply]
`$TERM` is empty, because vim is not running in a terminal. Please provide concrete reproduction steps

**Reply by Wanpaku** [reply]
Reproduction steps: 
1.Install Manjaro OS 24.0-6, 
2. Install gVim 9.1.0648-1. 
3. Launch gVim. 
4. Enter the command ":echo $TERM".

**Reply by chrisbra** [reply]
I mean for the scrolling issue. As I said `$TERM` is expected to be empty

**Reply by Wanpaku** [reply]
Reproduction steps:
1.Install Manjaro OS 24.0-6,
2. Install gVim 9.1.0648-1.
3. Launch gVim.
4. Write some program in C++ with long output (simple cycle from 0 to 100 will be enough).
5. Try to scroll upward to see beginning of the terminal output. It doesn't scroll upward in any way(with keys or mouse).

**Reply by chrisbra** [reply]
That are no reproduction steps. 
Start with `gvim --clean`. Run `:term` call your program. You can use `seq 1 100` to generate a lot of lines.
So something like this:
```
:term seq 1 100
<try to scroll with the mouse> or using `<C-B>/<C-F>` or using `j/k`
```

**Reply by chrisbra** [reply]
Note: if you are using the mouse, make sure to have the mouse pointer over the terminal window in order to scroll. Tested this one, and it seems to work as expected.

**Comment by mgedmin** [comment]
Note that if you press `Ctrl-\` twice, it will send a SIGQUIT signal to the program running in the terminal.  This is one of the reasons why I prefer `Ctrl-W capital-N` instead of `Ctrl-\ Ctrl-N` for entering terminal normal mode.
